### ISSUE TITLE
SR: Track Distance: The visual tabs are announced as buttons and without a state

### ISSUE THREAD

**Issue by nbhardwaj-del** [Expensify/App#74848]
## Action Performed
Prerequisites:
- The user is signed

1. Using Chrome + JAWS, open the URL - https://new.expensify.com/
2. Navigate to the "+" icon and press Enter
3. Tab to the "Track distance' menu item and press Enter to open the Pop-up
4. Tab to the 'Map and Manual' tabs and observe the screen reader announcement.

Other Occurrences:
1. On Create Expense: all three tabs
2. On Chat flow

## Expected Result
The controls "Map and Manual" should be announced with correct role as tabs and with the selected state. When focused, the screen reader should announce: "Map, Tab, 1 of 2, selected" (for the active tab) or "Manual, Tab, 2 of 2" (for the inactive tab).

## Actual Result
The controls used to switch between the "Map" and the "Manual" controls are announced with the incorrect role of button instead of a tab. Also, the state of the active control (the one currently selected and displaying content) is not being announced. The user cannot programmatically determine if they are currently viewing the "Map" content or the "Manual" content and the incorrect ole announcement may lead to misrepresentation of controls behavior.

HTML:
<div class="css-g5y9jx r-18u37iz r-kzbkwu r-u9wvl5"><div class="css-g5y9jx" id="1300" style="opacity: 1; flex-grow: 1;"><button aria-label="Map" role="button" data-selection-scrapper-hidden-element="true" data-tag="pressable" tabindex="0" class="css-g5y9jx r-1otgn73 r-1loqt21 r-1awozwy r-18c69zk r-18u37iz r-eu3ka r-1777fci r-xyw6el r-9ffhgg r-lrvibr" data-testid="distanceMap" type="button" data-component="GenericPressable" data-element="Pressable" data-source-file="BaseGenericPressable.tsx" style="background-color: rgb(230, 225, 218);"><div class="css-g5y9jx"><div class="css-g5y9jx" style="opacity: 0;"><div class="css-g5y9jx" data-testid=""><svg xmlns="[http://www.w3.org/2000/svg"](http://www.w3.org/2000/svg%22) xml:space="preserve" viewBox="0 0 20 20" width="16" height="16" hovered="false" pressed="false" fill="#A2A9A3"><path d="M6 2.5.684 4.272A1 1 0 0 0 0 5.221v10.892a1 1 0 0 0 1.316.949L6 15.5zm1 0 6 2v13l-6-2zm13 1.387a1 1 0 0 0-1.316-.949L14 4.5v13l5.316-1.772a1 1 0 0 0 .684-.949z"></path></svg></div></div><div class="css-g5y9jx r-1p0dtai r-1d2f490 r-u8s1d r-zchlnj r-ipm5af" style="opacity: 1;"><div class="css-g5y9jx" data-testid=""><svg xmlns="[http://www.w3.org/2000/svg"](http://www.w3.org/2000/svg%22) xml:space="preserve" viewBox="0 0 20 20" width="16" height="16" hovered="false" pressed="false" fill="#03D47C"><path d="M6 2.5.684 4.272A1 1 0 0 0 0 5.221v10.892a1 1 0 0 0 1.316.949L6 15.5zm1 0 6 2v13l-6-2zm13 1.387a1 1 0 0 0-1.316-.949L14 4.5v13l5.316-1.772a1 1 0 0 0 .684-.949z"></path></svg></div></div></div><div class="css-g5y9jx"><div class="css-g5y9jx" style="opacity: 1;"><div dir="auto" class="css-146c3p1" style="color: rgb(0, 46, 34); font-size: 13px; text-align: left; font-family: &quot;Expensify Neue&quot;, &quot;Windows Segoe UI Emoji&quot;, &quot;Noto Color Emoji&quot;, &quot;Custom Emoji Font&quot;; font-style: normal; font-weight: 700; margin-left: 8px; line-height: 18px;">Map</div></div><div class="css-g5y9jx r-1p0dtai r-1d2f490 r-u8s1d r-zchlnj r-ipm5af" style="opacity: 0;"><div dir="auto" class="css-146c3p1" style="color: rgb(118, 132, 126); font-size: 13px; text-align: left; font-family: &quot;Expensify Neue&quot;, &quot;Windows Segoe UI Emoji&quot;, &quot;Noto Color Emoji&quot;, &quot;Custom Emoji Font&quot;; font-style:

## Area issue was found
Track Distance

## Failed Checkpoint
WCAG 4.1.2 Name, Role, Value
https://www.w3.org/WAI/WCAG22/Understanding/name-role-value

## User Impact
Providing role, state, and value information on all user interface components enables compatibility with assistive technology, such as screen readers, screen magnifiers, and speech recognition software, used by people with disabilities.

## Suggested Resolution
Ensure the correct role and state are announced for the controls by wraping the "Map" and "Manual" controls in an element with role="tablist" and apply role="tab" to both the "Map" and "Manual" controls (instead of role="button"). To announce the selected state use "aria-selected" attribute and set its value to true.

Refer:
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles

## Environment(s)
Windows Windows 11 Home - Chrome

## Screenshots/Videos
https://github.com/user-attachments/assets/61db5b4c-a2ee-4b22-a272-2d1fbc072c0b

<img width="1919" height="962" alt="Image" src="https://github.com/user-attachments/assets/2fbaa9c5-e166-4b45-bdb0-b41389b504d4" />

**Version Number:** NA
**Reproducible in staging?:** NA
**Reproducible in production?:** Yes
**Email or phone of affected tester (no customers):** NA
**Logs:** https://stackoverflow.com/c/expensify/questions/4856 NA
**Notes/Photos/Videos:** Any additional supporting documentation Yes
**Expensify/Expensify Issue URL:** https://platform.applause.com/company/12102/products/36605/community-issues/6996747
**Issue reported by:** NA
**Slack conversation:** NA

**Comment by mallenexpensify** [comment]
`Accessibility` issue, will be addressed soon.
